### PR TITLE
Add option mapping with defaults

### PR DESCRIPTION
**Pull request by mackermans**
Proposal for option mapping.

I believe that this way it's easier to distinguish between `file`, `json` and `string` input (defaulting to file input). Output is either `json` or `string`. 

(I thought `nullInput` wasn't descriptive enough)

Todo in this PR:
- [x] Add option documentation to README
- [x] Add more meaningful tests

Optional: Add more options as you see fit ([jq-manual](https://stedolan.github.io/jq/manual/))

### CODE REVIEW

**Comment by mackermans** [comment]
Related to #2

File: src/jq.js [hunk 1]
```diff
@@ -23,7 +17,11 @@ export const run = (filter, json, options = {}) => {
     const { cmd, params } = createJqCommand(filter, json, options)
     execa(cmd, params)
     .then(({ stdout }) => {
-      return resolve(JSON.parse(stdout))
+      if (options.output === 'json') {
```
Comment (davesnx) [inline src/jq.js]: we should always return the stdout instead of adding the options here

File: src/options.js [hunk 2]
```diff
@@ -1,6 +1,51 @@
...
-    `${JSON.stringify(json)} | ${filter}`
-  ]
+import { validateJsonPath } from './utils'
+
+const optionDefaults = {
+  input: 'file',
+  output: 'string'
+}
+
+const optionMap = {
```
Comment (davesnx) [inline src/options.js]: why not a Map(), or a Array, brah?

File: test/jq.test.js [hunk 3]
```diff
@@ -12,10 +12,11 @@ const asteriskPathFixture = path.join(ROOT_PATH, 'src', '*.js')
 const filter = '. | map(select(.a == .id))'
 
 describe('jq core', () => {
-  it('should return a stdout object', () => {
-    return expect(
-      run(filter, jsonPathFixture)
-    ).to.eventually.be.instanceof(Object)
+  it('should return an stdout object', (done) => {
```
Comment (davesnx) [inline test/jq.test.js]: buuuuu, don't do the done!

File: src/options.js [hunk 4]
```diff
@@ -1,6 +1,51 @@
...
+const mergeOptionDefaults = (options = {}) => {
+  Object.keys(optionDefaults).forEach((key) => {
+    if (options[key] === undefined) {
+      options[key] = optionDefaults[key]
+    }
+  })
+}
+
+export const parseOptions = (filter, json, options = {}) => {
+  mergeOptionDefaults(options)
```
Comment (davesnx) [inline src/options.js]:
```js
Object.assign({}, defaultOptions, options)
```

**Comment by mackermans** [comment]
@davesnx `Object.assign` is only available in ES6 and therefore fails in node 0.12. If we want to use that method we have to add the `babel-polyfill` dependency to runtime... 

I think we should revert to the former `mergeOptionDefaults` method. What do you think?

File: src/options.js (hunk 2 above)
Comment (mackermans) [inline src/options.js]: What do you mean?

File: src/jq.js (hunk 1 above)
Comment (mackermans) [inline src/jq.js]: I understand your point but I think it's good to have the option to modify the output... how do you think we should do it, if not here?

**Comment by davesnx** [comment]
Make sense @mackermans, leave the mergeProps

**Comment by mackermans** [comment]
@davesnx Can you review this PR and merge if you agree?

**Pull request merged**